src/changelog.ts

```diff
@@ -192,42 +187,137 @@ function stringifyLinkReferenceDefinitions(
+ * @returns A unreleased link reference definition string.
+ */
+function getUnreleasedLinkReferenceDefinition(
+  repoUrl: string,
+  tagPrefix: string,
+  releases: ReleaseMetadata[],
+  versionBeforePkgRename?: string,
+  tagPrefixBeforePkgRename?: string,
+): string {
+  // A list of release versions in descending SemVer order
+  const descendingSemverVersions = releases
+    .map(({ version }) => version)
+    .sort((a: Version, b: Version) => {
+      return semver.gt(a, b) ? -1 : 1;
+    });
+  const latestSemverVersion = descendingSemverVersions[0];
+  const hasReleases = descendingSemverVersions.length > 0;
+  // if there is a package renamed, the tag prefix before the rename will be considered for compare
+  // [Unreleased]: https://github.com/ExampleUsernameOrOrganization/ExampleRepository/compare/test@0.0.2...HEAD
+  const tagPrefixToCompare =
+    tagPrefixBeforePkgRename && versionBeforePkgRename === latestSemverVersion
+      ? tagPrefixBeforePkgRename
+      : tagPrefix;
+
+  return `[${unreleased}]: ${
+    hasReleases
+      ? getCompareUrl(
+          repoUrl,
+          `${tagPrefixToCompare}${latestSemverVersion}`,
+          'HEAD',
+        )
+      : withTrailingSlash(repoUrl)
+  }`;
+}
+
+/**
+ * Get a list of release link reference definitions.
+ *
+ * @param repoUrl - The URL for the GitHub repository.
+ * @param tagPrefix - The prefix used in tags before the version number.
+ * @param releases - The releases to generate link definitions for.
+ * @param [versionBeforePkgRename] - A version string of the package before being renamed.
+ * @param [tagPrefixBeforePkgRename] - A tag prefix string of the package before being renamed.
+ * @returns A list of release link reference definitions.
+ */
+function getReleaseLinkReferenceDefinitions(
+  repoUrl: string,
+  tagPrefix: string,
+  releases: ReleaseMetadata[],
+  versionBeforePkgRename?: string,
+  tagPrefixBeforePkgRename?: string,
+): string[] {
+  const chronologicalVersions = releases.map(({ version }) => version);
+  let tagPrefixToCompare = tagPrefix;
+  const releaseLinkReferenceDefinitions = releases.map(({ version }) => {
+    let diffUrl;
+    // once the version matches with original version, rest of the lines in changelog will be assumed as migrated tags
+    if (tagPrefixBeforePkgRename && versionBeforePkgRename === version) {
+      tagPrefixToCompare = tagPrefixBeforePkgRename;
+    }
+    if (version === chronologicalVersions[chronologicalVersions.length - 1]) {
+      diffUrl = getTagUrl(repoUrl, `${tagPrefixToCompare}${version}`);
+    } else {
+      const versionIndex = chronologicalVersions.indexOf(version);
+      const previousVersion = chronologicalVersions
+        .slice(versionIndex)
+        .find((releaseVersion: Version) => {
+          return semver.gt(version, releaseVersion);
+        });
+      // when there is a package renamed below if will fix the validation for renamed package's first release
```

Comment:
This comment is slightly confusing. Do you mean the first release of the package as a whole, or just the original name of the package?

Reply:
>renamed package's first release
the first release after the package has been renamed. I'll update it.